industry/foc/float/foc_ident.c

```diff
@@ -343,6 +347,127 @@ int foc_ident_ind_run_f32(FAR struct foc_ident_f32_s *ident,
   return ret;
 }
 
+#ifdef CONFIG_INDUSTRY_FOC_IDENT_FLUX
+/****************************************************************************
+ * Name: foc_ident_fluxlink_run_f32
+ *
+ * Description:
+ *   Run flux linkage identification routine
+ *
+ * Input Parameter:
+ *   ident - pointer to FOC ident routine
+ *   in    - pointer to FOC routine input data
+ *   out   - pointer to FOC routine output data
+ *
+ ****************************************************************************/
+
+int foc_ident_fluxlink_run_f32(FAR struct foc_ident_f32_s *ident,
+                               FAR struct foc_routine_in_f32_s *in,
+                               FAR struct foc_routine_out_f32_s *out)
+{
+  int   ret  = FOC_ROUTINE_RUN_NOTDONE;
+  float curr_avg  = 0.0f;
+  float volt_avg  = 0.0f;
+  static float dir    = DIR_CW;
+  static float vq_sum = 0.0f;
+  static float vd_sum = 0.0f;
+  static float iq_sum = 0.0f;
+  static float id_sum = 0.0f;
+  static float vel = 0.0f;
+  static int aveg_sum  = 0;
```

Comment:
why use static here

Reply:
to get the average value by accumulation

Reply:
vel will pass to ramp function to get and maintain open loop velocity,

Reply:
see my previous comment